File: benchmark_phepcong/benchmarch_add.cpp
#include<stdio.h>
#include<time.h>
#include<math.h>

int tinhtong (int N){
    int total=0;
    for (int j=0;j<N;j++)
        {
            total+=j;
        }
    return total;

}
float nhieuPhepToan(int N){
    float result=0.0;
    for (int j=1;j<=N;j++)
    {
        result +=(float)j/(float)((j+1)+(j+2));
        
    }
    return result;
}
void fibonaci_generation(int N)
{
    int first =1;
    int second =1 ;
    printf ("%d ",first);
    printf(" %d ",second);
    int third =0;
    while (third <N)
    {
        third=first+second;
        printf("%d ",third);
        first=second;
        second=third;
    }



}
int main ()
{
    //printf("hello this is vinh");
    int max_rand, min_rand;
    int count;
    double time_tinhtong, time_nhieupheptinh,time_fibonaci;
    printf(" nhap so luong so N can test : ");
    scanf("%d", &count);
    printf("nhap gioi han tren :");
    scanf("%d", &max_rand);
    printf("nhap gioi han duoi :");
    scanf("%d",&min_rand);
    for (int i=0 ;i<count; i++)

    { 
        int randNum =min_rand+rand() %(max_rand-min_rand+1);

        int total =0;
        
        clock_t time_start=clock();
        total =tinhtong(randNum);
        printf("tong day so la %d \n",total);
        time_tinhtong+=(double)(clock() - time_start)/CLOCKS_PER_SEC;
        printf("Time taken: %.5fs\n", time_tinhtong);
        

        float result=0;
        time_start=clock();
        result =nhieuPhepToan(randNum);
        printf("ket qua thuc hien nhieu phep toan la %f \n",result);
        time_nhieupheptinh+=(double)(clock() - time_start)/CLOCKS_PER_SEC;
        printf("Time taken: %.5fs\n", time_nhieupheptinh);

        time_start=clock();
        printf("day fibonaci nho hon %d \n",randNum);
        fibonaci_generation(randNum);
        time_fibonaci+=(double)(clock() - time_start)/CLOCKS_PER_SEC;
        printf("Time taken: %.5fs\n", time_fibonaci);

    }  
    printf("thoi gian tinh tong trung binh: %.5fs \n",time_tinhtong/count);
    printf("thoi gian thuc hien nhieu phep tinh trung binh :%.5fs\n",time_nhieupheptinh/count);
    printf("thoi gian thuc hien tao day fibonaci : %.5fs \n",time_fibonaci/count);
    printf("ket thuc chuong trinh"); 
    return 0 ;

}

File: benchmark_phepcong/BenchmarchAdd.java
import java.util.Random;
import java.util.Scanner;

import javax.swing.plaf.synth.SynthScrollBarUI;



public class BenchmarchAdd {

   /* This is my first java program.
    * This will print 'Hello World' as the output
    */

   public static void main(String []args) {
      Scanner scanner=new Scanner(System.in);
      System.out.println("nhap gioi han tren cua day random ");
      int rand_max=scanner.nextInt();
      System.out.println("nhap so lan thuc hien :");
      int run_counter=scanner.nextInt();
      double runningTimeTinhTong=0;
      double runningTimeNhieuPhepTinh=0;
      double runningTimeFibonaci=0;

      for( int i=0;i<run_counter;i++){
         Random rand= new Random(); 
         int rand_int=rand.nextInt(rand_max);
         System.out.println(rand_int);

         double startTime=System.nanoTime();
         tinhtong(rand_int);
         double runningTime =System.nanoTime()-startTime;
         double runningTime_second=(double)runningTime/1000000000.0;
         runningTimeTinhTong+=runningTime_second;     
         //System.out.println("this is running time in second "+ runningTime_second);
         
         startTime=System.nanoTime();
         nhieupheptinh(rand_int);
         runningTime =System.nanoTime()-startTime;
         runningTime_second=(double)runningTime/1000000000.0;
         runningTimeNhieuPhepTinh+=runningTime_second;
         //System.out.println("this is running time in second "+ runningTime_second);

         startTime=System.nanoTime();
         fibonaci_generate(rand_int);
         runningTime =System.nanoTime()-startTime;
         runningTime_second=(double)runningTime/1000000000.0;
         runningTimeFibonaci+=runningTime_second;       
         //System.out.println("this is running time in second "+ runningTime_second);
      }

      System.out.println("\n thoi gian chay trung binh tinh tong "+runningTimeTinhTong/run_counter);
      System.out.println("thoi gian chay trung binh nhieu phep tinh :"+runningTimeNhieuPhepTinh/run_counter );
      System.out.println("thoi gian chay trung binh tao day fibonaci :"+runningTimeFibonaci/run_counter);
   }
 
   public static void tinhtong(int n){
      int tong=0;
      for (int i=0;i<n;i++){
         tong +=i;
      }
      System.out.println("ket qua cua tong day so la "+tong);
   }
   public static void nhieupheptinh(int n ){
      float result=0;
      for (int i =0;i<n;i++)
      {
         result += (float)i/(float)((i+1)+(i+2));

      }
      System.out.println("ket qua cua nhieu phep tinh la :"+result);
   }
   public static void fibonaci_generate(int n){
      int first =1;
      int second =1;
      int third =0;
      System.out.print(first+" "+second );
      while(third < n){
         third=first +second;
         System.out.print(" "+third);
         first=second;
         second=third;
         
      }    
   }
}
